### PR TITLE
Allow users to configure whether Lantern should set the system proxy

### PR DESCRIPTION
**Pull request by myleshorton**

### CODE REVIEW

**Comment by myleshorton** [comment]
Hey @oxtoacart you mind taking a look at this one?

**Comment by myleshorton** [comment]
Or @atavism you want to give this one a look?

**Comment by oxtoacart** [comment]
Sure, I'm kind of in this area for the Windows stuff anyway.

File: src/github.com/getlantern/flashlight/main/settings.go [hunk 1]
```diff
@@ -73,91 +72,114 @@ func Load(version, revisionDate, buildDate string) {
...
+// SetSystemProxy sets whether or not to set system proxy when lantern starts
+func (s *Settings) SetSystemProxy(enable bool) {
+	s.Lock()
+	defer s.Unlock()
+	s.SystemProxy = enable
+	if enable {
+		go pacOn()
+	} else {
+		go pacOff()
 	}
```
Comment (oxtoacart) [inline src/github.com/getlantern/flashlight/main/settings.go]: @myleshorton It's very unlikely to happen in practice, but if this function is called in rapid succession, there's no guarantee that these goroutines will run in the right order.  Better would be to set up a chan bool and a single goroutine that toggles based on what it gets from that channel.

File: src/github.com/getlantern/flashlight/main/settings.go [hunk 2]
```diff
@@ -73,91 +72,114 @@ func Load(version, revisionDate, buildDate string) {
...
+
+// SetSystemProxy sets whether or not to set system proxy when lantern starts
+func (s *Settings) SetSystemProxy(enable bool) {
+	s.Lock()
+	defer s.Unlock()
+	s.SystemProxy = enable
+	if enable {
+		pacOn()
+	} else {
+		pacOff()
```
Comment (oxtoacart) [inline src/github.com/getlantern/flashlight/main/settings.go]: The handler for updating this from the UI already uses a buffered channel, so the simplest thing to do to ensure consistent ordering is to just not spawn goroutines for pacOn and pacOff.

**Comment by oxtoacart** [comment]
LGTM!  I just made one small tweak to not launch goroutines for the pac settings change.

**Comment by oxtoacart** [comment]
Once travis check passes, I'll merge.

**Comment by oxtoacart** [comment]
@myleshorton Sorry, I just remembered that I need to integrate this with the ui.lantern.io change.  Working on that now, will submit a PR to the PR.

**Comment by myleshorton** [comment]
Awesome thanks @oxtoacart! The other thing we need is to add logic to not
set the UI to ui.lantern.io if the user does not have Lantern set as the
system proxy.

**Pull request merged**